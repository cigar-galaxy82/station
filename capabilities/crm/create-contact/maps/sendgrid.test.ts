import { SuperfaceTest } from '@superfaceai/testing-lib';

describe(`crm/create-contact/sendgrid`, () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest();
  });

  describe('CreateContact', () => {
    it('should return error if email is missing', async () => {
      await expect(
        superface.run({
          profile: 'crm/create-contact',
          provider: 'sendgrid',
          useCase: 'CreateContact',
          input: {},
        })
      ).resolves.toMatchSnapshot();
    });

    it('should perform successfully', async () => {
      await expect(
        superface.run({
          profile: 'crm/create-contact',
          provider: 'sendgrid',
          useCase: 'CreateContact',
          input: {
            email: 'TEST@example.com',
            customProperties: {
              e1_T: 'value',
            },
          },
        })
      ).resolves.toMatchSnapshot();
    });
  });
});
