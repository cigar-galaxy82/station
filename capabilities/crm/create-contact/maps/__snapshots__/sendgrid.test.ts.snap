// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`crm/create-contact/sendgrid CreateContact should perform successfully 1`] = `
Object {
  "value": Object {
    "id": null,
  },
}
`;

exports[`crm/create-contact/sendgrid CreateContact should return error if email is missing 1`] = `
Object {
  "error": "HTTPError: Expected HTTP error
AST Path: definitions[0].statements[0]
Original Map Location: Line 5, column 3",
}
`;
